packages/core/crates/core-misc/src/environment.rs

```diff
@@ -0,0 +1,300 @@
+        let protocol_config_path = format!("{}/packages/core/protocol-config.json", mono_repo_path);
+
+        let data = real::read_file(protocol_config_path.as_str())?;
+
+        let mut protocol_config = ok_or!(serde_json::from_slice::<ProtocolConfig>(&data))?;
+
+        for (id, env) in protocol_config.environments.iter_mut() {
+            env.id = id.to_owned();
+        }
+
+        for (id, network) in protocol_config.networks.iter_mut() {
+            network.id = id.to_owned();
+        }
+
+        Ok(protocol_config)
+    }
+}
+
+impl ProtocolConfig {
+    /// Returns a list of environments which the node is able to work with
+    fn supported_environments(&self, mono_repo_path: &str) -> JsResult<Vec<Environment>> {
+        let version =
+            PackageJsonFile::from_json_file(&mono_repo_path).and_then(|c| c.coerced_version())?;
+
+        let mut allowed: Vec<Environment> = vec![];
+
+        for (_, env) in self.environments.iter() {
+            let range = env.version_range.to_owned();
+
+            if satisfies(version.to_owned(), range).unwrap() {
+                allowed.push(env.to_owned())
+            }
+        }
+
+        Ok(allowed)
+    }
+}
+
+#[derive(Deserialize)]
+struct PackageJsonFile {
+    version: String,
+}
+
+impl FromJsonFile for PackageJsonFile {
+    fn from_json_file(mono_repo_path: &str) -> JsResult<Self> {
+        let package_json_path = format!("{}/packages/hoprd/package.json", mono_repo_path);
+
+        let data = real::read_file(package_json_path.as_str())?;
+
+        ok_or!(serde_json::from_slice::<PackageJsonFile>(&data))
+    }
+}
+
+impl PackageJsonFile {
+    fn coerced_version(&self) -> JsResult<String> {
+        /*
+         * Coerced full version using
+         * coerce_version('42.6.7.9.3-alpha') // '42.6.7'
+         */
+        coerce_version(self.version.to_owned())
+    }
+}
+
+#[derive(Serialize, Clone)]
+pub struct ResolvedEnvironment {
+    /// the environment identifier, e.g. monte_rosa
+    pub id: String,
+    pub network: NetworkOptions,
+    pub environment_type: EnvironmentType,
+    /// an Ethereum address
+    pub channels_contract_address: String,
+    pub channel_contract_deploy_block: u32,
+    /// an Ethereum address
+    pub token_contract_address: String,
+    /// an Ethereum address
+    pub xhopr_contract_address: String,
+    /// an Ethereum address
+    pub boost_contract_address: String,
+    /// an Ethereum address
+    pub stake_contract_address: String,
+    /// an Ethereum address
+    pub network_registry_proxy_contract_address: String,
+    /// an Ethereum address
+    pub network_registry_contract_address: String,
+}
+
+impl ResolvedEnvironment {
+    /// Returns the environment details, returns an error if environment is not supported
+    fn new(
+        mono_repo_path: &str,
+        environment_id: &str,
+        maybe_custom_provider: Option<&str>,
+    ) -> JsResult<Self> {
```

Comment:
This is non-wasm bound code that returns a WASM bindgen specific type (`Result<T,JsValue>`). Please do not use wasm types in pure Rust code (= outside `mod wasm`)

Reply:
changed, as discussed